rest/permissions.py

```diff
@@ -0,0 +1,46 @@
+"""
+Permission classes for REST views
+"""
+
+from __future__ import unicode_literals
+
+from rest_framework.permissions import (
+    BasePermission,
+    SAFE_METHODS,
+)
+from django.http import Http404
+
+from learningresources.models import Repository
+from learningresources.api import (
+    get_repo,
+    PermissionDenied,
+    NotFound,
+)
+
+
+# pylint: disable=protected-access
+class AddRepoPermission(BasePermission):
```

Comment:
Could the `ViewRepoPermission` class below be used to validate view access, freeing up `AddRepoPermission` for checking for insert permission?